docs/platform/connector-development/connector-builder-ui/authentication.md

```diff
@@ -93,57 +97,128 @@ curl -X GET \
    - Response parsing configuration for access token and expiration
-   - Support for custom parameters and headers
 
-3. **Optional Refresh Token URL Configuration** - Defines how to refresh expired access tokens:
+3. **Optional Refresh Token Configuration** - Defines how to refresh expired access tokens:
    - Refresh token endpoint URL (if different from access token URL)
-   - Parameter placement for client credentials and refresh token
-   - Custom header and body configurations
+   - Custom parameter placement for client credentials and refresh token
+   - Configurable header and body parameters
 
-To learn more about the Declarative OAuth 2.0 add see a variety of example implementations please refer to the [Declarative OAuth 2.0](/platform/connector-development/config-based/advanced-topics/oauth) documentation.
+#### Configuration Options
 
+- **Client ID Key** - Custom field name for client_id parameter (defaults to "client_id")
+- **Client Secret Key** - Custom field name for client_secret parameter (defaults to "client_secret")
+- **Authorization Code Key** - Custom field name for authorization code parameter (defaults to "code")
+- **Redirect URI Key** - Custom field name for redirect_uri parameter (defaults to "redirect_uri")
+- **Scopes** - Space-separated list of OAuth scopes to request
+- **Access Token Key** - Field name in response containing the access token (defaults to "access_token")
+- **Access Token Headers** - Custom headers to include in access token requests
+- **Access Token Parameters** - Custom query parameters for access token requests
 
-#### Partial OAuth (legacy)
+#### Example Configuration
 
-The partial OAuth flow is a simpler flow that allows you to use an existing access (or refresh) token to authenticate with the API.
+For an API that uses standard OAuth 2.0 with custom parameter names:
 
+- **Consent URL**: `https://api.example.com/oauth/authorize?client_id={{client_id_value}}&redirect_uri={{redirect_uri_value}}&scope={{scope_value}}&state={{state_value}}`
+- **Access Token URL**: `https://api.example.com/oauth/token`
+- **Scopes**: `read write`
 
-The catch is that the user needs to implement the start of the OAuth flow manually to obtain the `access_token` and the `refresh_token` and pass them to the connector as a part of the `config` object.
+When a user authenticates, they'll be redirected to the consent URL to grant permissions, then the connector will automatically exchange the authorization code for an access token.
 
+To learn more about Declarative OAuth 2.0 and see additional example implementations, please refer to the [Declarative OAuth 2.0](/platform/connector-development/config-based/advanced-topics/oauth) documentation.
 
-The OAuth authentication method implements authentication using an OAuth2.0 flow with a [refresh token grant type](https://oauth.net/2/grant-types/refresh-token/) and [client credentials grant type](https://oauth.net/2/grant-types/client-credentials/).
+### OAuth (Legacy)
```

Comment:
This section should be combined with the above based on what I mentioned earlier